giles-eco/src/main/java/edu/asu/diging/gilesecosystem/web/core/apps/impl/RegisteredAppDatabaseClient.java

```diff
@@ -3,23 +3,26 @@
 import java.util.List;
 
 import javax.persistence.EntityManager;
-import javax.persistence.PersistenceContext;
-import javax.persistence.TypedQuery;
 
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
-import org.springframework.transaction.annotation.Transactional;
 
 import edu.asu.diging.gilesecosystem.util.exceptions.UnstorableObjectException;
 import edu.asu.diging.gilesecosystem.util.store.objectdb.DatabaseClient;
 import edu.asu.diging.gilesecosystem.web.core.apps.IRegisteredApp;
 import edu.asu.diging.gilesecosystem.web.core.apps.IRegisteredAppDatabaseClient;
+import edu.asu.diging.gilesecosystem.web.core.apps.RegisteredAppRepository;
 
-@Transactional
 @Component
 public class RegisteredAppDatabaseClient extends DatabaseClient<IRegisteredApp> implements IRegisteredAppDatabaseClient {
-
-    @PersistenceContext(unitName="entityManagerFactory")
-    private EntityManager em;
+    
+    @Autowired
+    private final RegisteredAppRepository registeredAppRepository;
+    
+    @Autowired
+    public RegisteredAppDatabaseClient(RegisteredAppRepository registeredAppRepository) {
```

Comment:
since `RegisteredAppRepository ` is autowired above, we don't need this contructor